nextcloud/android
# Optimize and fix some bugs for autoupload

## Fixes
- Makes Autoupload file scan faster by checking for changes since last file system scan and not query database for last known change timestamp (saves the database query) 
- Fixes logic mistake to start foreground service when doing big scan and not for content observer events
- Fixes start of infinite ContentObservers 

### To test performance improvements
- Select a big custom folder for autoupload
- wait till all files are scheduled for upload (activate pause to not upload all the files) 
- In uploads, tab swipe down to restart file sync job
- Check how long the "preparing for upload" notification stays there
- Compare to version before https://github.com/nextcloud/android/pull/12719 since the foreground notification was broken with it

### To test if autoupload still working
- Add document (not video or image) to one of the autoupload folder 
- Start file sync worker by swiping down in uploads tab
- Wait till "prepare for upload" notification is gone
- Check if file was scheduled for upload

### Maybe also test
- Upgrade (Database changes)
- ContentObserver changed file detection by adding image to auto upload folder (should add to upload within 30s)
